### PR TITLE
fix ceilometer and aodh api (to use apache)

### PR DESCRIPTION
**Pull request by Xiaohua-Shen**
and some small fix for log.

### CODE REVIEW

File: roles/aodh/defaults/main.yml [hunk 1]
```diff
@@ -23,14 +23,27 @@ aodh:
...
       - { name: python-ceilometerclient }
-    system_dependencies: []
+    system_dependencies: 
+      ubuntu:
+        - apache2
+        - libapache2-mod-wsgi
+      rhel:
+        - httpd
+        - mod_wsgi
+        - mod_proxy_uwsgi
```
Comment (zfeldstein) [inline roles/aodh/defaults/main.yml]: Is this needed? looks like we should be ok with just installing mod_wsgi

File: library/systemd_service.py [hunk 2]
```diff
@@ -129,6 +129,8 @@ def main():
             if not changed:
                 module.exit_json(changed=False, result="ok")
             else:
+                cmd_stop = 'systemctl stop %s.service' % module.params['name']
+                os.system(cmd_stop)
```
Comment (zfeldstein) [inline library/systemd_service.py]: We use this module for installing and configuring systemd unit files. I would recommend using the ansible service module if you want to stop/start services.

File: library/upstart_service.py [hunk 3]
```diff
@@ -92,6 +92,8 @@ def main():
             if not changed:
                 module.exit_json(changed=False, result="ok")
             else:
+                cmd_stop = 'service %s stop' % module.params['name']
+                os.system(cmd_stop)
```
Comment (zfeldstein) [inline library/upstart_service.py]: Same comment as above. Use ansible service module.

File: roles/aodh/defaults/main.yml [hunk 4]
```diff
@@ -23,14 +23,27 @@ aodh:
       - { name: PyMySQL }
       - { name: python-memcached }
       - { name: python-ceilometerclient }
-    system_dependencies: []
+    system_dependencies: 
+      ubuntu:
+        - apache2
```
Comment (zfeldstein) [inline roles/aodh/defaults/main.yml]: apache2 will get installed by the apache role, as you're dep'ing on it in meta/main.yml

**Comment by zfeldstein** [comment]
Using the mod_wsgi approach and having apache run the python app we no longer have the ability to just restart the aodh service when changes are required. This also opens up the possibility if something is broken with aodh it could prevent apache from starting, which would cause keystone to go down (and horizon). I would recommend using the uwsgi approach in the same fashion that we currently setup keystone.

**Comment by epwn** [comment]
ok to test

**Comment by Xiaohua-Shen** [comment]
based on community document:
1. suggest to use mod_proxy_uwsgi for keystone: https://github.com/openstack/keystone/blob/master/doc/source/apache-httpd.rst,  the sample configuration is 5 processes and 1 thread for each process.
2. suggest to use mod_wsgi for ceilometer:
https://docs.openstack.org/project-install-guide/telemetry/newton/install-base-rdo.html#configure-the-apache-http-server.  the sample configuration is 2 processes and 10 threads for each process.

For my understanding, use mod_proxy_uwsgi and standalone uwsgi service has good isolation for each app, but may add more overhead for performance (since it has one more proxy pass beween apache and uwsgi service).
For this PR,  I still use mod_wsgi.  And I will raise another PR using mod_proxy_uwsgi to compare.

**Comment by Xiaohua-Shen** [comment]
retest

**Comment by nirajdp76** [comment]
@Xiaohua-Shen let's close this PR.

**Pull request closed without merging**